command/build.go

```diff
@@ -188,33 +214,26 @@ func (c *BuildCommand) Run(args []string) int {
 				artifacts.m[name] = runArtifacts
 				artifacts.Unlock()
 			}
-		}(b)
+		}()
 
 		if cfgDebug {
 			log.Printf("Debug enabled, so waiting for build to finish: %s", b.Name())
 			wg.Wait()
 		}
 
-		if !cfgParallel {
+		if cfgParallelBuilds == 1 {
 			log.Printf("Parallelization disabled, waiting for build to finish: %s", b.Name())
 			wg.Wait()
 		}
 
-		if interrupted {
-			log.Println("Interrupted, not going to start any more builds.")
-			break
-		}
 	}
 
 	// Wait for both the builds to complete and the interrupt handler,
 	// if it is interrupted.
 	log.Printf("Waiting on builds to complete...")
 	wg.Wait()
 
-	log.Printf("Builds completed. Waiting on interrupt barrier...")
-	interruptWg.Wait()
```

Comment:
I may be missing something but I really don't understand how this interruptWg could unlock before ????